lib/domain/scooter_state.dart

```diff
@@ -16,8 +17,20 @@ enum ScooterState {
   linking,
   disconnected;
 
-  static ScooterState fromString(String stateString) {
-    switch (stateString) {
+  static ScooterState fromStateAndPowerState(String state, ScooterPowerState powerState) {
+    if (powerState == ScooterPowerState.hibernating) {
+      return ScooterState.hibernating;
+    }
+
+    if (powerState == ScooterPowerState.hibernatingImminent) {
+      return ScooterState.hibernatingImminent;
+    }
+
+    if (state == "off" && powerState == ScooterPowerState.booting) {
```

Comment:
This is the actual fix by checking that the scooter state is `off`. Only then we consider the power state.